README.md

```diff
@@ -412,14 +412,56 @@ DeploymentApi api = new DeploymentApi(client);
 
 For more customization, creating a [HeaderProvider](https://sap.github.io/cloud-sdk/docs/java/features/connectivity/http-destinations#about-headerproviders) is also possible.
 
-## Requirements and Setup
+## Requirements and Setup for AI Core
 
-### Set AI Core credentials as environment variable
+For any AI Core service interaction, the SAP Cloud SDK for AI requires credentials to be available at application runtime.
+By default, the credentials are extracted automatically from a service instance of type "aicore" bound to the application. 
+Running the application locally without such service binding will throw an exception:
 
-- Running the application locally without a service binding will throw:
+```
+Could not find any matching service bindings for service identifier 'aicore'
+```
+
+There are multiple options to register the service binding:
+* Regular service binding in SAP BTP Cloud Foundry (resulting in `VCAP_SERVICES` env var entry).
+* Set an environment variable explicitly: `AICORE_SERVICE_KEY`
+* Define and use a _Destination_ in _BTP Destination Service_.
+* Leveraging `"user-provided"` service binding (currently not recommended).
+* Define and use a custom `ServiceBinding` or `ServiceBindingAccessor` declaration in application (not recommended).
+
+### Regular service binding in SAP BTP Cloud Foundry
+
+* In the _SAP BTP Cockpit_ navigate to the target application.
+  Go to _Service Bindings_ and click _Bind Service_.
+  In the wizard create or bind an existing service instance of type `aicore`.
+* (Or) use `manifest.yaml` to add the service binding declaratively.
+* (Or) use the CF CLI to add the service binding via command line.
+
+<details><summary>After application restart, there should be an "aicore" entry in environment variable <code>VCAP_SERVICES</code>.</summary>
+
+```json
+{
+  "aicore": [
+      {
+        "clientid": "...",
+        "clientsecret": "...",
+        "url": "...",
+        "identityzone": "...",
+        "identityzoneid": "...",
+        "appname": "...",
+        "serviceurls": {
+          "AI_API_URL": "..."
+        }
+      }
+  ]
+}
+```
+
+</details>
+
+### Set credentials as dedicated environment variable
```

Comment:
Mention that this is recommended if you need to access an external AI Core service instance or for local use

Reply:
Currently the README is already almost unreadable with all the details and sample code. I don't want to explain and motivate reasons for the options to resolve a service binding.
Not only because of complexity and readability, but also any documented "reasoning" is subject to change over time. I'm certain we will see changes to best-practices, once the project gains traction.